module-2/backend/week_two.md

```diff
@@ -26,20 +35,72 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+a =  Team.find(4)
+a.owner
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+https://www.dropbox.com/s/7k3ly3b6mqqtkyn/Screenshot%202017-08-27%2023.40.37.png?dl=0
+
 6. Define foreign key, primary key, and schema.
+
+foreign key - is another tables data column inserted into another table
+primary key - is that colum on the other table
+schema - is how your database knows what is what. the "schematic" for which your database knows what goes where when importing or storing data.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+owner_id is the foreign key on the team table through id on the owner table
+
 8. What are the parts of an HTTP response?
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+
+group - takes an argument of a colum on a table to group the data together by the column
+joins - joins takes an argument of another table to make them into a even larger table so you can manipulate both tables at once
+count - turns the previous query into a hash with the key of whatever you gave it previously and the value of the number of items in that column 
+find - finds the row of the id you pass it
+maximum - finds the maximum value in the column you pass it
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+
+rake db:drop - drops your database tables
+rake db:create - creates your database tables
+rake db:migrate - pushes your migrations into your schema
+
 3. What two columns does `t.timestamps null: false` create in our database?
+
+created at and updated at
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+
+one to many
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+
+a teacher will belongs_to :school
+and a school will has_many :teachers
+https://www.dropbox.com/s/ox07fk1hq83z6t8/Capture.PNG?dl=0
+
 6. Give an example of when you might want to store information besides ids on a join table.
+
+not sure on this one
```

Comment:
Think about something like `invoice_items` where we store a quantity and unit price. Or potentially `courses` might be a join between students and teachers and it could store a student's grade for that course.